CruCentralCoast/Events/EventsVC.swift

```diff
@@ -23,11 +34,38 @@ class EventsVC: UITableViewController {
         self.tableView.estimatedRowHeight = 140
         
         DatabaseManager.instance.subscribeToDatabaseUpdates(self)
-        self.dataArray = DatabaseManager.instance.getEvents()
     }
-
+    
+    override func viewWillAppear(_ animated: Bool) {
+        let subscribedMovements = LocalStorage.preferences.getObject(forKey: .subscribedMovements) as? [String] ?? []
+        self.databaseArray = DatabaseManager.instance.getEvents()
+        
+        //reset the dataArray because I am appending it below.
+        self.dataArray.removeAll()
+        
+        self.databaseArray.forEach { (event) in
+            event.movements.forEach({ (movement) in
+                //check if the movement is in the subscribed movements and the dataArray does not already have the event to avoid repeats.
+                if subscribedMovements.contains(movement.id) && !self.dataArray.contains(event){
+                    self.dataArray.append(event)
+                    //return since at least one movement from the event is in the subscribed movements.
+                    return
+                }
+            })
+        }
+        
+        //reload the data in the tableview.
```

Comment:
This comment is probably unnecessary since the following line reads the same thing

Reply:
haha good point!